source/external/recast/CMakeLists.txt

```diff
@@ -2,27 +2,34 @@ cmake_minimum_required(VERSION 3.6)
 project(recast)
 
 set(RECAST_SRC
-DebugUtils/DebugDraw.cpp
-DebugUtils/DetourDebugDraw.cpp
-Detour/DetourAlloc.cpp
-Detour/DetourCommon.cpp
-Detour/DetourNavMesh.cpp
-Detour/DetourNavMeshBuilder.cpp
-Detour/DetourNavMeshQuery.cpp
-Detour/DetourNode.cpp
-DetourCrowd/DetourCrowd.cpp
-DetourCrowd/DetourLocalBoundary.cpp
-DetourCrowd/DetourObstacleAvoidance.cpp
-DetourCrowd/DetourPathCorridor.cpp
-DetourCrowd/DetourPathQueue.cpp
-DetourCrowd/DetourProximityGrid.cpp
-DetourTileCache/DetourTileCache.cpp
-DetourTileCache/DetourTileCacheBuilder.cpp
-fastlz/fastlz.c
+    DebugUtils/DebugDraw.cpp
+    DebugUtils/DetourDebugDraw.cpp
+    Detour/DetourAlloc.cpp
+    Detour/DetourCommon.cpp
+    Detour/DetourNavMesh.cpp
+    Detour/DetourNavMeshBuilder.cpp
+    Detour/DetourNavMeshQuery.cpp
+    Detour/DetourNode.cpp
+    DetourCrowd/DetourCrowd.cpp
+    DetourCrowd/DetourLocalBoundary.cpp
+    DetourCrowd/DetourObstacleAvoidance.cpp
+    DetourCrowd/DetourPathCorridor.cpp
+    DetourCrowd/DetourPathQueue.cpp
+    DetourCrowd/DetourProximityGrid.cpp
+    DetourTileCache/DetourTileCache.cpp
+    DetourTileCache/DetourTileCacheBuilder.cpp
+    fastlz/fastlz.c
 )
 
 add_library(recast STATIC
```

Comment:
Please remove STATIC

Reply:
as far as I am confirmed,  no need to do it, add STATIC is clear since we build static library only.

https://cmake.org/cmake/help/v3.0/command/add_library.html

Reply:
Without STATIC you can build both STATIC and SHARED.
Both OK. We can agree that we always put STATIC (and SHARED when needed: android, lgpl lib)